src/std/misc/list.ss

```diff
@@ -210,6 +210,6 @@ package: std/misc
 ;; a non-empty list, otherwise an empty list is returned.
 (defrules when-list-or-empty ()
   ((_ list body body* ...)
-   (if (null? list)
+   (if (not (pair? list))
      []
      body body* ...)))
```

Comment:
there is also a bug here, the body should be wrapped in a begin:
```
(begin body body* ...)
```

Note: I didn't write this macro!